src/Generation/MethodDetails.php

```diff
@@ -115,7 +115,17 @@ private static function maybeCreatePaginated(ServiceDetails $svc, MethodDescript
         $pageSize = $inputMsg->desc->getFieldByName('page_size');
         $pageToken = $inputMsg->desc->getFieldByName('page_token');
         $nextPageToken = $outputMsg->desc->getFieldByName('next_page_token');
-        $resources = $outputMsg->desc->getFieldByNumber(1);
+        // Find the resoures field. Although AIP states that the resource field should be field number 1,
```

Comment:
Is this for the response message? If so, using the first repeated field handled all protos we came across in Java.

Curious: Will the repeated resource be needed for other parts of the generated code? If so (as in the case in Java), should this be extracted to a common parsing location?

Reply:
I'll leave it as first-by-number-and-position for now, as I think that's mentioned in AIP and I think what C# does. This can always be updated if it needs to be more flexible.

So far, the repeated resource is not required anywhere else, but yes it should be exposed publicly on this class if that becomes the case. If I understand your question correctly, then this is the common parsing location, but at the moment the result isn't made available publicly (as it's not currently required).